cli/conf.py

```diff
@@ -74,17 +74,35 @@ def parse_args(cls, module_name, config, args=None):
 
         :param module_name: the module name: installer|provisioner|tester
         """
-        cmd = clg.CommandLine(cls._get_specs(module_name, config))
+
+        clg_options, global_opts, subcommand_opts = \
+            SpecLoader.load(module_name, config)
+
+        cmd = clg.CommandLine(clg_options)
         res_args = vars(cmd.parse(args))
 
-        # always load default values for command0
-        default_file = os.path.join(
-            config.get('defaults', 'settings'),
-            module_name,
-            res_args['command0'],
-            DEFAULT_INI
-        )
-        defaults = IniFileType(default_file)[res_args['command0']]
+        # get the default values
+        command_args = subcommand_opts[res_args['command0']]
+        defaults = {k: v.get('default', None)
+                    for k, v in command_args.iteritems()}
+
+        # generate config file if required
+        if res_args['generate-conf-file']:
+            try:
+                file_name = res_args['generate-conf-file']
+                out_config = ConfigParser.ConfigParser()
+
+                section = res_args['command0']
+                out_config.add_section(section)
+                for opt, value in defaults.iteritems():
+                    if value is not None:
+                        out_config.set(section, opt, value)
+                with open(file_name, 'w') as configfile:    # save
+                    out_config.write(configfile)
+            except Exception as ex:
+                raise exceptions.IRException(ex.message)
+            finally:
+                sys.exit(0)
```

Comment:
I don't think we should have sys.exit anywhere here.
I suggest setting `dry-run=True` and replacing output file with the rest of the the output.
Alternatively, generate-conf-file can be a new subcommand, with "virsh" as its input.